### PR TITLE
Feature/modul 318 amelioration m dialog

### PR DESCRIPTION
**Pull request by fredduguay**
## `@ulaval/modul-components`
# PR Checklist

- [x] Provide a small description of the changes introduced by this PR
Add multiple new props and behaviors to the dialog component (formerly called modal)

- New dialog types inheriting their visual behavior from the m-message component (similar to State), visual is toggled via prop.
- New optional secondary button added.
- Cancel Link is now optional.
- New optional hint inheriting it's behavior from the m-message component (new state: hint)
- Allow optional primary and secondary/precision labels to be specified for buttons.
- Allow optional cancelLink label.
- Rename onclick events for the buttons and cancelLink, change emitted value when events are called.
- New 'hint' state available in m-message and m-message-page components.
- [x] Include links to issues
https://jira.dti.ulaval.ca/browse/MODUL-318
- [ ] Openshift deployment requested

- [x] Include this section in the release notes
BREAKING CHANGE, multiple changes to this component, please see below.
----
- Add multiple new props and behaviors to the dialog component (formerly called modal)
- New dialog types inheriting their visual behavior from the m-message component (similar to State), visual is toggled via prop.
- New optional secondary button added.
- Cancel Link is now optional.
- New optional hint inheriting it's behavior from the m-message component (new state: hint)
- Allow optional primary and secondary/precision labels to be specified for buttons.
- Allow optional cancelLink label.
- Rename onclick events for the first button and cancelLink, change emitted value when events are called. onOk() is now onPrimaryButton(), emit = 'primary'. onCancel() is now onCancelLink(), emit = 'cancel'.
- New onSecondaryButton  event added, emit = 'secondary'.
- Props changed: 
okLabel ---> prBtnLabel.
okPrecision ---> prBtnPrecision
negativeLink --> cancelLink
cancelLink --> cancelLinkLabel

- [ ] Other info

### CODE REVIEW

**Comment by fredduguay** [comment]
@jipigi, @chuckmah Ceci est un nouveau PR, j'ai fermé https://github.com/ulaval/modul-components/pull/537 car j'ai renommé la branche 318 car le nom était trop long

File: src/components/dialog/dialog.ts [hunk 1]
```diff
@@ -26,13 +37,21 @@ export class MDialog extends ModulVue implements PortalMixinImpl {
     @Prop()
     public message: string;
     @Prop()
-    public okLabel: string | undefined;
+    public hint: string;
     @Prop()
-    public okPrecision: string | undefined;
+    public prBtnLabel: string | undefined;
     @Prop()
```
Comment (chuckmah) [inline src/components/dialog/dialog.ts]: J'enleverais le | undefined ca va faire des trucs weird quand on va générer les meta data

ie:
```
  @Prop()
    public okPrecision: string | undefined;	    public prBtnLabel: string ;
```

File: src/components/dialog/dialog.ts [hunk 2]
```diff
@@ -5,17 +5,28 @@ import { Prop } from 'vue-property-decorator';
...
 import { ModulVue } from '../../utils/vue/vue';
-import ButtonPlugin from '../button/button';
+import ButtonPlugin, { MButton } from '../button/button';
 import { DIALOG_NAME } from '../component-names';
 import I18nPlugin from '../i18n/i18n';
 import LinkPlugin from '../link/link';
+import MessagePlugin, { MMessageState } from '../message/message';
 import WithRender from './dialog.html?style=./dialog.scss';
+import { MPopup } from '../popup/popup';
+import { MInputStyle } from '../input-style/input-style';
```
Comment (chuckmah) [inline src/components/dialog/dialog.ts]: Elever les imports inutile

File: src/components/message/message.scss [hunk 3]
```diff
@@ -33,6 +33,23 @@ $m-message-size: 32px;
...
 
+    &.m--is-state-hint {
+        @include m-message-skin-default($m-color--hint);
+
+        border: 1px solid $m-color--success;
+
+        .m-message {
+            &__wrap {
+                font-style: italic;
+                font-size: 19px;
```
Comment (chuckmah) [inline src/components/message/message.scss]: pas de variable pour ca ? , ca aiderais den avoir une lorsqu'on va switcher en rem

**Pull request closed without merging**